int8[] int_array
float32[] float_array
string string_val
